upnext/create_party.py

```diff
@@ -15,16 +16,29 @@ def create_party_in_db(request, form):
 def create_playlist(request, party_name):
     username = 'up--next'
     user = request.user.social_auth.get(provider='spotify')
-    print user.extra_data['access_token'], '\n\n\n', user.extra_data['refresh_token']
     token_info = tokens.token_read()
     token = token_info['ACCESS_TOKEN']
     if token:
         sp = spotipy.Spotify(auth=token)
         sp.trace = False
         playlist = sp.user_playlist_create(username, party_name)
         party = Party.objects.get(party_name=party_name)
+        initial_uri = party.uri
         party.uri = playlist['uri']
         party.created_at = timezone.now()
         party.save()
+
+        # Load songs from initial playlist into new playlist
+        if initial_uri != "UpNext - Start a new playlist":
+            tracks_info = sp.user_playlist_tracks(user, playlist_id=initial_uri)["items"]
+            uris = map(lambda x: x["track"]["uri"].split(":")[-1], tracks_info)
+            track_titles = map(lambda x: x["track"]["name"], tracks_info)
+            artists = map(lambda x: x["track"]["artists"][0]["name"], tracks_info)
+            for i in range(len(uris)):
+                if not party.track_set.filter(uri=uris[i]).exists():
```

Comment:
Why do you need this check? Is it in case they have duplicate tracks in the playlist?